### PR TITLE
Use @mapbox/gazetteer for benchmark coordinates

### LINKED ISSUE #8059: Replace style_locations with @mapbox/gazetteer
Right now, there's a locally managed [`bench/lib/style_locations.js`](https://github.com/mapbox/mapbox-gl-js/blob/master/bench/lib/style_locations.js) file that is the mirrored version of style-benchmark-locations.json in [`@mapbox/gazetteer`](https://github.com/mapbox/gazetteer/blob/mb-pages/mapbox-streets/style-benchmark-locations.json). To keep everything in sync so that other platforms (native) are using the same location data in their benches we should replace what's here with that.

### PR DESCRIPTION
**Pull request by tristen**
Replace `bench/lib/style_locations.js` with the same place data made available in `@mapbox/gazetteer`. 

@ansis thanks for the tip to use `MercatorCoordinate.fromLngLat` to derive tile index. Could I get you to review this?

---
Closes #8059

### CODE REVIEW

**Comment by tristen** [comment]
🤔 It's not clear to me why tests are failing.

**Comment by tristen** [comment]
> The bench changes look good (aside from the test failures), but did you accidentally ran `yarn upgrade`?

@mourner hmm I don't _think_ I did? I just tried blowing away `node_modules/` + `yarn.lock`,  running `yarn` again and I don't any difference against the lock file :/

**Comment by ryanhamley** [comment]
You may need to just update Yarn. That fixed this issue for me.

**Comment by tristen** [comment]
> You may need to just update Yarn. That fixed this issue for me.

@ryanhamley that seemed promising but there are still no changes against `yarn.lock` after upgrading yarn to `1.15.2`.

**Comment by mourner** [comment]
@tristen do not delete the `yarn.lock` file. Run `yarn` when it's reset to its master state.

File: bench/lib/style_locations.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+/* eslint camelcase: 0 */
```
Comment (mourner) [inline bench/lib/style_locations.js]: Just do `feature.properties['place_name']` instead.

File: package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
   "dependencies": {
     "@mapbox/appropriate-images-react": "^1.0.0",
     "@mapbox/dr-ui": "0.6.0",
+    "@mapbox/gazetteer": "^3.1.0",
```
Comment (mourner) [inline package.json]: Needs to be in devDeps.

Comment (tristen) [reply package.json]: Oy ... me and yarn today :/

**Pull request merged**